File: Introduction/Java Implementation/Factorial.java
//Program to find factorial of a number n recursively
import java.util.Scanner;

class Factorial {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter the value of n: ");
        int n = sc.nextInt();
        System.out.println(n + "! = " + fact(n));   
    }

    private static int fact(int n){
        if(n == 0){
            return 1;
        }
        return n * fact(n - 1);
    }
}

File: Introduction/C implementation/factorial.c
//Program to find factorial of a number n recursively
#include<stdio.h>

int fact(int n);
int main() 
{
    int n;
    printf("Enter the value of n: ");
    scanf("%d", &n);
    printf("%d! = %d\n", n, fact(n));
    return 0;
}

int fact(int n) 
{
    if (n == 0) 
    {
        return 1;
    }
    return n * fact(n - 1);
}